internal/ethapi/api.go

```diff
@@ -987,17 +987,16 @@ func DoCall(ctx context.Context, b Backend, args TransactionArgs, blockNrOrHash
 	// Execute the message.
 	gp := new(core.GasPool).AddGas(math.MaxUint64)
 	result, err := core.ApplyMessage(evm, msg, gp)
+	if err != nil {
+		return result, fmt.Errorf("err: %w (supplied gas %d)", err, msg.Gas())
+	}
 	if err := vmError(); err != nil {
```

Comment:
I think there was nothing wrong with this, the way it was. Se https://go.dev/play/p/w3zIAvKsWbV 

The `err := vmError` is scoped, and does not overwrite the outer-scoped `err`.

Reply:
However, I agree that it makes sense to check the returned err first, like this PR does. So I still approve

Reply:
Actually, I'm not so sure. So the `vmErr` will (now) potentially return an underlying error, e.g if a part of the trie could not be loaded. 
In a lot of cases, this will also trigger a secondary error during precheck -- typically something will be zero which shouldn't have been. 

In this case, the `vmErr()` is the more interesting error, since it shows the underlying cause ("missing trie node") instead of the effect ("insufficient funds for transfer"). 

So now I'm leaning towards reverting this change

Reply:
Make sense, reverted the calling site.